cpp/oneapi/dal/table/backend/homogen_table_builder_impl.hpp

```diff
@@ -75,8 +75,20 @@ class homogen_table_builder_impl {
         column_count_ = 0;
     }
 
-    void set_feature_type(feature_type ft) {
-        throw dal::unimplemented(dal::detail::error_messages::method_not_implemented());
+    void set_feature_type(feature_type ft, std::int64_t idx) {
+        if (column_count_ == 0) {
+            throw dal::domain_error{ "column_count should be > 0 " };
+        }
+        ONEDAL_ASSERT(idx >= 0);
+        ONEDAL_ASSERT(idx < column_count_);
+
+        allocate_ftypes_if_need();
+        feature_type* ftypes_ptr = ftypes_.need_mutable_data().get_mutable_data();
+        ftypes_ptr[idx] = ft;
```

Comment:
`ftypes_` are not reset in `reset()` methods at least. Make sure that the build logic is not broken in other scenarios.